### PR TITLE
Add coq-serapi to OPAM.

### PR DESCRIPTION
**Pull request by ejgallego**
Coq SerAPI is a new toplevel and serialization system for Coq.

### CODE REVIEW

**Comment by rgrinberg** [comment]
Curious about the choice of packaging here. Why isn't the META part of the package itself?

**Comment by ejgallego** [comment]
> Curious about the choice of packaging here. Why isn't the META part of the package itself?

The META file is part of the Coq packaging, however it has some bugs that couldn't be fixed on time for 8.7.1 (https://github.com/coq/coq/pull/6447)

coq-serapi is the first user of the Coq's META file, and I wouldn't like to wait for 8.7.2 for it to work.

I agree this is not an optimal solution, however I have no way to add the fix to the Coq OPAM package itself and have SerAPI depend on the right coq version (which is doable in Debian), so this interim solution seemed like the best for me.

On the other hand, it is an ugly hack, so feel free to propose a different approach.

Basically, we would like to add the fixes in the above PR to Coq's 8.7.1.

**Comment by ejgallego** [comment]
Yup so indeed I am convinced now that this approach has quite a few problems, I'd rather update Coq's OPAM to include the fixes, right?

How can I express the dependency on the fixed Coq ?

**Comment by rgrinberg** [comment]
Well if you fix the 8.7.1 of coq, you'll just need to need add it as a lower bound on your package here (as far as I understand)

**Comment by ejgallego** [comment]
Yup, I will update Coq first.

**Comment by ejgallego** [comment]
Ok so this depends now on https://github.com/ocaml/opam-repository/pull/11054 , and doesn't play dirty tricks anymore.

**Comment by ejgallego** [comment]
It seems `ppx_import/deriving` is broken again :(

**Comment by ejgallego** [comment]
Ok, `ppx_import` got quickly fixed, let's see how this works now!

**Comment by mseri** [comment]
This looks good to me. I am not sure why there are those three failures on the V2.0 all distros test. I think we should merge it unless you have objections @rgrinberg

**Comment by ejgallego** [comment]
Thanks for the review @mseri

File: packages/coq-serapi/coq-serapi.8.7.1+0.4/opam [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "coq"          { >= "8.7.1+1" & < "8.8" }
+  "camlp5"
+  "ocamlfind"    { build }
+  "ocamlbuild"   { build }
+  "ppx_import"   { build & >= "1.4" }
+  "ppx_deriving" { build & >= "4.2.1" }
+  "cmdliner"
+  "sexplib"
+  "ppx_driver"   { build & >= "v0.10.1" }
+  "ppx_sexp_conv"
```
Comment (rgrinberg) [inline packages/coq-serapi/coq-serapi.8.7.1+0.4/opam]: This should be a build dep as well

Comment (ejgallego) [reply packages/coq-serapi/coq-serapi.8.7.1+0.4/opam]: Indeed thanks, updated my local copy, I was not sure if the converted code was depending on some conversion-specific libraries, but I see all of them are in `Sexplib`.

**Comment by ejgallego** [comment]
Thanks a lot to all who have helped! It took 18 months to finally get to the point that this could be released in OPAM.
